src/app.js

```diff
@@ -1,20 +1,87 @@
-const express = require('express');
+require("dotenv").config();
+
+const express = require("express");
+const session = require("express-session");
 const app = express();
-const cors = require('cors')
-const favicon = require('express-favicon');
-const logger = require('morgan');
 
-const mainRouter = require('./routes/mainRouter.js');
+const morgan = require("morgan");
+const bodyParser = require("body-parser");
+const cors = require("cors");
+const favicon = require("express-favicon");
+const { xss } = require("express-xss-sanitizer");
+const helmet = require("helmet");
+
+const passport = require("passport");
+const mongoose = require("mongoose");
+mongoose.set("strictQuery", true);
+const asyncErrors = require("express-async-errors");
+const cookieParser = require("cookie-parser");
+const rateLimit = require("express-rate-limit");
+const limiter = rateLimit({
+  windowMs: 15 * 60 * 1000, // 15 minutes
+  max: 60, // limit each IP to 100 requests per windowMs
+});
+const logger = require("./logger");
+
+// middleware setup
+
+app.use(xss());
 
-// middleware
-app.use(cors());
 app.use(express.json());
-app.use(express.urlencoded({ extended: false }));
-app.use(logger('dev'));
-app.use(express.static('public'))
-app.use(favicon(__dirname + '/public/favicon.ico'));
+//Security middleware
+app.use(helmet());
+
+app.use(cors({ origin: [/localhost:8000$/], credentials: true }));
```

Comment:
Is `8000` the port that the frontend will run on? We want to tell our backend, which urls from the front to allow through and not block with CORS

Reply:
the  frontend repo runs on 3000 port as mentioned in README. Thus should be changed for 3000?

Reply:
Yes please change it to 3000 :+1:

Reply:
Changed to 3000 , commit -4682f9f48cd7941d2c850db797dcede94158b5de